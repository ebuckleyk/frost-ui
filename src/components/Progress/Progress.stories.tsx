import * as React from 'react';
import { Meta, StoryObj } from '@storybook/react';

import { Progress } from './Progress';

function ProgressDemo() {
  const [progress, setProgress] = React.useState(13);

  React.useEffect(() => {
    const timer = setTimeout(() => setProgress(66), 500);
    return () => clearTimeout(timer);
  }, []);

  return (
    <div className="min-w-[150px]">
      <Progress value={progress} className="w-[100%]" />
    </div>
  );
}
type ComponentType = React.ComponentProps<typeof Progress>;
const meta: Meta<ComponentType> = {
  component: Progress,
  render: ProgressDemo,
};

export default meta;

type Story = StoryObj<ComponentType>;
export const Demo: Story = {};
