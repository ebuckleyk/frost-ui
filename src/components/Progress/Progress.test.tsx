import * as React from 'react';
import { render } from '@testing-library/react';

import { Progress } from './Progress';

function Component() {
  const [progress] = React.useState(13);

  return (
    <div className="min-w-[150px]">
      <Progress value={progress} className="w-full" />
    </div>
  );
}

describe('Progress', () => {
  it('should render Progress component to match snapshot', () => {
    const result = render(<Component />);
    expect(result).toMatchSnapshot();
  });
});
