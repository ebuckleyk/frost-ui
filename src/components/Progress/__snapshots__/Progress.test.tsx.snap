// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Progress > should render Progress component to match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="min-w-[150px]"
      >
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          class="relative h-2 overflow-hidden rounded-full bg-primary/20 w-[100%]"
          data-max="100"
          data-state="indeterminate"
          role="progressbar"
        >
          <div
            class="h-full w-full flex-1 bg-primary transition-all"
            data-max="100"
            data-state="indeterminate"
            style="transform: translateX(-87%);"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="min-w-[150px]"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        class="relative h-2 overflow-hidden rounded-full bg-primary/20 w-[100%]"
        data-max="100"
        data-state="indeterminate"
        role="progressbar"
      >
        <div
          class="h-full w-full flex-1 bg-primary transition-all"
          data-max="100"
          data-state="indeterminate"
          style="transform: translateX(-87%);"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
